### PR TITLE
JDF 159 - Demonstrate usage of Deactivateable from Deltaspike updating relevant Quickstarts that uses Extension 

### PR DESCRIPTION
**Pull request by rafabene**

### CODE REVIEW

**Comment by rafabene** [comment]
The Deactivateable feature was added to cdi-portable-extension and cdi-veto quickstarts. But I'm now wondering if we should keep just on cdi-portable-extension to avoid any confusion that could exist mixing a veto with deactivateable. What you think?

**Comment by LightGuard** [comment]
I don't think we want it on every CDI extension, just the ones using DeltaSpike. Pulling in DS for two classes in an already small extension seems overkill to me. The idea is they demonstrate one concept, so we shouldn't needlessly be confusing the users.

**Comment by rafabene** [comment]
Ok. So I'll revert the cdi-veto quickstart and keep on cdi-portable-extension only. Thanks

**Comment by rafabene** [comment]
Will create another QS that shows how to deactivate an existing DeltaSpike extension

File: deltaspike-deactivatable/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Author: Rafael Benevides
+Level: Beginner
+Technologies: CDI, DeltaSpike
+Summary: Demonstrate usage of Deactivateable.
+Target Product: WFK
+
+What is it?
+-----------
+
+DeltaSpike artifacts can be deactivated manually (e.g. to improve the performance if a part isn't needed, to provide a custom implementation if the default implementation isn't pluggable by default or to bypass an implementation which causes an issue).
```
Comment (pmuir) [inline deltaspike-deactivatable/README.md]: As we are demonstrating this facility, we should say "DeltaSpike allows..."

File: deltaspike-deactivatable/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+To deactivate a class it's required to implement ClassDeactivator. Returning 'false' or 'true' allows to de-/activate the class in question. This project has a ClassDeactivator that deactivates the ExcludeExtension. In this case the `MyBean` is annotated with `@Exclude` but continues to get injected.
```
Comment (pmuir) [inline deltaspike-deactivatable/README.md]: s/it's required/you need to/

Normally it's "true or false" :-)

Spell out activate or deactivate

"In this case the `MyBean` is annotated with `@Exclude` but continues to get injected." -> "If the `ExcludeExtension` is deactivated then `MyBean`, which is annotated with `@Exclude`, won't be excluded!"

File: deltaspike-deactivatable/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+3. Type the following command to run the test goal with the following profile activated:
+
+        mvn clean test -Parq-jbossas-remote
+
+
+Run tests from JBDS
+-----------------------
+
+To be able to run the tests from JBDS, first set the active Maven profile in project properties to be either 'arq-jbossas-managed' for running on
+managed server or 'arq-jbossas-remote' for running on remote server.
```
Comment (pmuir) [inline deltaspike-deactivatable/README.md]: Make this one line :-)

File: deltaspike-deactivatable/pom.xml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+            <groupId>org.jboss.spec.javax.ejb</groupId>
+            <artifactId>jboss-ejb-api_3.1_spec</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <!-- Deltaspike API. we use compile scope as we need its API -->
+        <dependency>
+            <groupId>org.apache.deltaspike.core</groupId>
+            <artifactId>deltaspike-core-api</artifactId>
+            <scope>compile</scope>
```
Comment (pmuir) [inline deltaspike-deactivatable/pom.xml]: Compile scope is default, no need to specify it,

File: deltaspike-deactivatable/pom.xml [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+        </dependency>
+
+        <!-- Deltaspike Impl. we use runtime scope as we its implementation 
+            dependencies only on runtime -->
+        <dependency>
+            <groupId>org.apache.deltaspike.core</groupId>
+            <artifactId>deltaspike-core-impl</artifactId>
+            <scope>compile</scope>
```
Comment (pmuir) [inline deltaspike-deactivatable/pom.xml]: This should be runtime scope

**Comment by pmuir** [comment]
Looks good beyond the minor issues commented on.

File: deltaspike-deactivatable/README.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
+DeltaSpike Deactivateable: Demonstrate usage of Deactivateable
```
Comment (sgilda) [inline deltaspike-deactivatable/README.md]: Use quickstart name before colon. 
s/deactivateable/deactivatable

For example:
deltaspike-deactivatable: Demonstrate use of DeltaSpike Deactivatable

File: deltaspike-deactivatable/README.md (hunk 2 above)
Comment (rafabene) [inline deltaspike-deactivatable/README.md]: We need to suggest the use of "true or false" to DS doc also: https://cwiki.apache.org/confluence/display/DeltaSpike/Temporary+Documentation#TemporaryDocumentation-Deactivatable

:smile:

File: deltaspike-deactivatable/pom.xml (hunk 5 above)
Comment (rafabene) [inline deltaspike-deactivatable/pom.xml]: The extension that we need to deactivate is on this dependency. I put a note about it.

**Comment by sgilda** [comment]
There are a few minor typos in the README file and pom.xml files where `deactivatable` is mistyped as 'deactivateable`.

README typos:
1. In the header - deltaspike-deactivatable: Demonstrate usage of Deactivateable
2. Summary: Demonstrate usage of Deactivateable

pom.xml typos:
1. <name>JBoss AS Quickstarts: DeltaSpike Deactivateable</name>
2. <description>DeltaSpike Deactivateable: Demonstrate usage of Deactivateable</description>

Other than that, this looks good to me.

**Comment by sgilda** [comment]
Looks good to me if @LightGuard  and @pmuir  are OK with this one.

**Comment by pmuir** [comment]
I'm happy :-)

**Comment by sgilda** [comment]
Rebased upstream, squashed commits, and merged.

**Pull request closed without merging**